### DISCUSSION TITLE
Proposal: Conditional operator inside condition

### DISCUSSION THREAD

**Discussion by mmaderic** [dotnet/csharplang#2697] [Language Ideas] [answered]
Hi,

I've encountered situation where this might be lovely syntax sugar if possible.
Enable conditioning conditinal operator.

For example I have condition statement where I would compare two equal items either with <= or < operators depending on some other condition.

Current solution is:
```
if (someCondition
                    ? (itemA < itemB)
                    : (itemA <= itemB))
                {
                    // do something
                }
```

Proposed solution
```
if(itemA (condition ? < : <=) itemB)
{
   // do something
}
```

**Comment by quinmars** [comment]
```csharp
if (itemA < itemB || (!someCondition && itemA == itemB))
{
}
```
That's in my eyes more readable: "Go on, if itemA is smaller than itemB. If some condition is not fulfilled, we also accept equality of itemA and itemB."

**Comment by MgSam** [comment, marked as answer]
Sounds like a very niche use case. I suggest just making a method or a custom `IEqualityComparer` if you're doing this frequently.

**Comment by mmaderic** [comment]
I agree it might be something we are mostly used to, but I had to think for a second about your solution to verify its corectness instead having instant readability which ternary operators might provide. .. Im fan of ternary, maybe thats why :)  and we are excluding double itemA and itemB specification.
